source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1704 @@
+    G (a , is-non-inc) = to-subtype-＝ is-non-inc-is-predicate sup-P-is-a
+     where
+      P : 𝓟 {𝓥} B
+      P = subset-of-small-ϕ-closed-subset
+           (non-inc-points-to-small-ϕ-closed-subsets (a , is-non-inc))
+      sup-P : ⟨ L ⟩
+      sup-P = point-non-inc-points
+               (small-ϕ-closed-subsets-to-non-inc-points
+                (non-inc-points-to-small-ϕ-closed-subsets (a , is-non-inc)))
+      sup-P-is-a : sup-P ＝ a
+      sup-P-is-a = is-supᴮ' a ⁻¹
+    is-qinv : qinv small-ϕ-closed-subsets-to-non-inc-points
+    is-qinv = (non-inc-points-to-small-ϕ-closed-subsets , H , G)
+
+\end{code}
+
+Using the previously displayed correspondance we can show that, under certain
+smallness assumptions on the least closed subset 𝓘nd ϕ, the monotone operator
+Γ ϕ has a least fixed point.
+
+\begin{code}
+
+  module small-𝓘nd-from-exists
+          (ind-e : inductively-generated-subset-exists L β h ϕ)
+         where
+
+   open inductively-generated-subset-exists ind-e
+   open trunc-ind-def L β h ϕ ind-e
+
+   module smallness-assumption (j : (b : B) → (b ∈ 𝓘nd) is 𝓥 small) where
+
+    private
+     𝓘' : (b : B) →  𝓥  ̇
+     𝓘' b = (resized (b ∈ 𝓘nd)) (j b) 
+
+     𝓘'-equiv-𝓘nd : (b : B) → 𝓘' b ≃ b ∈ 𝓘nd 
+     𝓘'-equiv-𝓘nd b = resizing-condition (j b)
+
+     𝓘'-to-𝓘nd : (b : B) → 𝓘' b → b ∈ 𝓘nd
+     𝓘'-to-𝓘nd b = ⌜ 𝓘'-equiv-𝓘nd b ⌝
+
+     𝓘nd-to-𝓘' : (b : B) → b ∈ 𝓘nd → 𝓘' b
+     𝓘nd-to-𝓘' b = ⌜ 𝓘'-equiv-𝓘nd b ⌝⁻¹
+
+     𝓘'-is-prop-valued : {b : B} → is-prop (𝓘' b)
+     𝓘'-is-prop-valued {b} = equiv-to-prop (𝓘'-equiv-𝓘nd b) (Ind-trunc b)
+
+     𝓘'-subset : 𝓟 {𝓥} B
+     𝓘'-subset = λ b → (𝓘' b , 𝓘'-is-prop-valued)
+
+     𝓘'-is-c-closed : (U : 𝓟 {𝓥} B)
+                    → U ⊆ 𝓘'-subset
+                    → (b : B) → b ≤ᴮ (⋁ 【 β , U 】)
+                    → b ∈ 𝓘'-subset
+     𝓘'-is-c-closed U C b o =
+       𝓘nd-to-𝓘' b (𝓘nd-is-c-closed U (λ x → 𝓘'-to-𝓘nd x ∘ C x) b o)
+      
+     𝓘'-is-ϕ-closed : (a : ⟨ L ⟩)
+                    → (b : B)
+                    → (b , a) ∈ ϕ
+                    → ((b' : B) → b' ≤ᴮ a → b' ∈ 𝓘'-subset)
+                    → b ∈ 𝓘'-subset
+     𝓘'-is-ϕ-closed a b p f =
+      𝓘nd-to-𝓘' b (𝓘nd-is-ϕ-closed a b p (λ b' → 𝓘'-to-𝓘nd b' ∘ f b'))
+
+     total-space-𝓘-is-small : (𝕋 𝓘nd) is 𝓥 small
+     total-space-𝓘-is-small = (𝕋 𝓘'-subset , Σ-cong λ b → 𝓘'-equiv-𝓘nd b)
+   
+     e : 𝕋 𝓘'-subset ≃ 𝕋 𝓘nd
+     e = resizing-condition total-space-𝓘-is-small
+
+     sup-𝓘 : ⟨ L ⟩
+     sup-𝓘 = ⋁ (𝕋 𝓘'-subset , β ∘ 𝕋-to-carrier 𝓘nd ∘ ⌜ e ⌝)
+
+     sup-𝓘-is-lub : (sup-𝓘 is-lub-of (𝕋 𝓘nd , β ∘ 𝕋-to-carrier 𝓘nd)) holds
+     sup-𝓘-is-lub = sup-of-small-fam-is-lub L (β ∘ 𝕋-to-carrier 𝓘nd)
+                                             total-space-𝓘-is-small
+
+\end{code}
+
+Note for Tom: I need to explore generalizing my notation since 𝓘nd isn't in 𝓥.
+
+\begin{code}
+
+    Γ-has-least-fixed-point : has-least-fixed-point L (Γ ϕ i)
+    Γ-has-least-fixed-point =
```

Comment:
I'd suggest splitting up this code into two parts: that `sup-𝓘` is a fixed point of `Γ ϕ i`, and that it is the least.